### PR TITLE
Fill GAMEINFO for name and password

### LINKED ISSUE #260: Game name missing in multiplayer
image

The game should display both the game name (host player name afaik) and possible password (the second line is omitted if no password is set):
image

This is most likely because the relevant buffers aren't being populated (some where in SourceX) on game start rather then a drawing issue.

### PR DESCRIPTION
**Pull request by mmaret**
Those information could be accessed by pressing tab key in multiplayer
Fix #260

Capture d’écran du 2019-09-26 23-02-47

Server Game name will be 0.0.0.0 or the address in the .ini file

### CODE REVIEW

File: SourceX/storm/storm_net.cpp [hunk 1]
```diff
@@ -111,12 +126,20 @@ BOOL SNetCreateGame(const char *pszGameName, const char *pszGamePassword, const
...
 	SRegLoadString("dvlnet", "bindaddr", 0, addrstr, 128);
+	strncpy(gpszGameName, addrstr, sizeof(gpszGameName) - 1);
+	if (pszGamePassword)
+		strncpy(gpszGamePassword, pszGamePassword, sizeof(gpszGamePassword) - 1);
 	*playerID = dvlnet_inst->create(addrstr, pszGamePassword);
 	return *playerID != -1;
 }
 
-BOOL SNetJoinGame(int id, char *pszGameName, char *pszGamePassword, char *playerName, char *userStats, int *playerID)
+BOOL SNetJoinGame(int id, char *pszGameName, char *pszGamePassword, char *playerName,
```
Comment (qndel) [inline SourceX/storm/storm_net.cpp]: why break in 2 lines? :P other than that, looks good to me

Comment (mmaret) [reply SourceX/storm/storm_net.cpp]: I've run clang-format -i SourceX/storm/storm_net.cpp -style=file on the file to be sure to match the project format.
I was thinking that this modification came from that, but, in fact, it's from my own tools and it's just fine for clang-format
Anyway, I'll remove it :)

File: SourceX/storm/storm_net.cpp [hunk 2]
```diff
@@ -69,6 +71,19 @@ BOOL SNetDropPlayer(int playerid, DWORD flags)
 
 BOOL SNetGetGameInfo(int type, void *dst, unsigned int length, unsigned int *byteswritten)
 {
+	switch (type) {
+	case GAMEINFO_NAME:
+		strncpy((char *)dst, gpszGameName, length);
```
Comment (AJenbo) [inline SourceX/storm/storm_net.cpp]: I probably would use `snprintf` instead as `strncpy` doesn't null terminate the string so a user providing a 128 char name might be able to break the client.

**Comment by AJenbo** [comment]
Very nice, it's really great to see all the small issues getting fixed :)

**Pull request merged**